oxen/tests/test_image_classification_loader.py

```diff
@@ -0,0 +1,65 @@
+import os
+import pytest
+import torch
+import tensorflow as tf
+from tensorflow import keras
+import oxen
+import numpy as np
+from oxen import Dataset
+from oxen.loaders import ImageClassificationLoader
+from torch.utils.data import TensorDataset, DataLoader
+
+def test_image_classification_dataloader_local(
+    celeba_local_repo_fully_committed, empty_local_dir
+):
+
+    repo = celeba_local_repo_fully_committed
+
+    train_file = os.path.join(repo.path, "annotations", "train.csv")
+    label_file = os.path.join(repo.path, "annotations", "labels.txt")
+
+    loader = ImageClassificationLoader(
+        imagery_root_dir = repo.path, 
+        label_file = label_file,
+        csv_file = train_file, 
+        path_name = "file", 
+        label_name = "hair_color"
+    )
+    data, labels, mapper = loader.run()   
+
+    assert data.shape == (5, 218, 178, 3), "Data not returned in expected shape"
+    assert labels.shape == (5,)
+    assert len(mapper.items()) == 3
+
+
+    # Test ease of use with pytorch
+    torch_data = TensorDataset(torch.from_numpy(data), torch.from_numpy(labels))
```

Comment:
We could also have class that is a torch.Dataset, that has __getitem__ and __len__ https://pytorch.org/tutorials/beginner/basics/data_tutorial.html

But this is good for now too

Reply:
made a ticket 👍